mkversion.sh

```diff
@@ -32,6 +32,12 @@ if [ ! -z "$1" ]; then
     o=shell
 fi
 
+# If the version is passed in the environment during build, let's use that
```

Comment:
I don't know if it's a good idea for mkversion to just be pulling random shell variables. I added the ability to pass it as a parameter so that you can control the input.